packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -59,42 +60,53 @@ async function createDataObject<TObj extends PureDataObject, I extends DataObjec
 
     // request mixin in
     runtimeClass = mixinRequestHandler(
-        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
-            (await PureDataObject.getDataObject(runtimeArg)).request(request),
-            runtimeClass);
+        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
+            const router: FluidObject<IFluidRouter> = (await runtimeArg.IFluidHandle?.get() as any);
+            if (router.IFluidRouter) {
+                return router.IFluidRouter.request(request);
+            }
+            return createResponseError(500, "Data store runtime handle is not an IFluidRouter", request);
```

Comment:
It feels like we should never get here. While I hate us using two different error flows, I feel like if we should never get here, assert is more appropriate, as it sends the right message (while 500 is "internal error", it may be read by users as bug on their side)

Reply:
I think you're right. Pre-PR the code assumes it always has an `IFluidRouter` and that shouldn't change. I'll update this.